Сделана авторизация, функционал полностью рабочий

korzhuk-xeniya opened the pull request.

Kseniia313 committed:
Добавлена авторизация пользователя

el-kg committed:
написан GlobalExceptionHandler

Поправлен выброс исключения ImageNotFoundException

исправлено название ImageNotFoundException

В ImageController поправлены обработки исключений

В AdsServiceImpl откорректирован выброс исключений

korzhuk-xeniya committed:
Initial commit

AlexandraGlinka committed:
тесты для UserMapper

AlexandraGlinka committed:
тесты для UserMapper FIX

AlexandraGlinka committed:
удален лишний код в пользователе, добавлено документирование сервиса

AlexandraGlinka committed:
тест для UserController (пока нерабочий)

korzhuk-xeniya committed:
поправлен логгер

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/2-feature-step-2-handlerException' into 3-feature-step-3

AlexandraGlinka committed:
UserServiceImplTest / getAuthUserInfoTest

AlexandraGlinka committed:
UserServiceImplTest / updateAuthUserInfoTest

AlexandraGlinka committed:
UserServiceImplTest / updateAvatarTest

korzhuk-xeniya committed:
покрыт тестами CommentMapper

Kseniia313 committed:
авторизация и регистрация пользователя

Kseniia313 committed:
авторизация и регистрация пользователя

Kseniia313 committed:
Merge remote-tracking branch 'origin/2-step-Kseniia-Bystrova' into 3-feature-step-3

# Conflicts:
#	pom.xml
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/controller/UserController.java
#	src/main/java/ru/skypro/homework/entity/User.java
#	src/main/java/ru/skypro/homework/repository/UserRepository.java
#	src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java
#	src/main/resources/liquibase/scripts/users.sql

Kseniia313 committed:
авторизация и регистрация пользователя

Kseniia313 committed:
авторизация и регистрация пользователя

Kseniia313 committed:
авторизация и регистрация пользователя

Kseniia313 committed:
авторизация и регистрация пользователя

Kseniia313 committed:
возвращены настройки БД в application.properties

Kseniia313 committed:
Merge branch '3-feature-step-3' into 3-step-Kseniia-B

Kseniia313 committed:
правки для мержа

Kseniia313 committed:
Merge pull request #7 from korzhuk-xeniya/3-step-Kseniia-B-work-branch

Авторизация и регистрация пользователя

Kseniia313 committed:
Поправила SQL запрос на создание таблицы users

Kseniia313 committed:
Merge remote-tracking branch 'origin/3-step-Kseniia-B-work-branch' into 3-feature-step-3

korzhuk-xeniya committed:
тесты для AdsMapper начало

korzhuk-xeniya committed:
правки в appication properties

korzhuk-xeniya committed:
AdsMapperTest done

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/3-feature-kxeny-tests' into 3-feature-step-3

AlexandraGlinka committed:
Merge remote-tracking branch 'origin/3-feature-user-tests-aglinka' into 3-feature-step-3

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/UserController.java
#	src/main/java/ru/skypro/homework/service/UserService.java
#	src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java

AlexandraGlinka committed:
UserServiceImpl - метод смены пароль заменен на метод с энкодером

korzhuk-xeniya committed:
CommentServiceImplTest в работе

korzhuk-xeniya committed:
CommentServiceImplTest в работе

OlgaLugacheva commented on pom.xml:
```diff
@@ -92,6 +92,14 @@
     </dependency>
+
+
+    <dependency>
+      <groupId>mysql</groupId>
+      <artifactId>mysql-connector-java</artifactId>
+      <version>8.0.23</version>
+    </dependency>
```
зачем это?

OlgaLugacheva commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -61,9 +62,14 @@ public ResponseEntity<AdsDto> getAllAds() {
+            return ResponseEntity.ok(adsMapper.adsToAdsDto(adsService.saveAd(createOrUpdateAdDto, authentication.getName(), file)));
+        } catch (Exception e){
+            log.error("Ошибка при добавлении объявления", e);
+            throw new RuntimeException("Ошибка при добавлении объявления", e);
```
ну давайте кастомные исключения, такое не очень бросать

OlgaLugacheva commented on src/main/java/ru/skypro/homework/controller/ImageController.java:
```diff
@@ -20,7 +20,12 @@ public class ImageController {
+           imageService.transferImageToResponse(id, response);
+       } catch(Exception e){
+           log.error("Ошибка при передаче изображения в ответ", e);
+           throw new RuntimeException("Ошибка при передаче изображения в ответ", e);
```
см выше

OlgaLugacheva commented on src/main/java/ru/skypro/homework/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,34 @@
+        // Логируем ошибку
+        log.error("An error occurred: {}", ex.getMessage(), ex);
+
+        // Возвращаем соответствующий статус ответа
```
для этого есть джава-доки) и стоит джава - доки в принципе дописать в те , методы которые не вполне очевидно , что они делают по названию

OlgaLugacheva commented on src/main/java/ru/skypro/homework/filter/SecurityUserService.java:
```diff
@@ -0,0 +1,76 @@
+        if (byEmail.isEmpty()) {
+            throw new UsernameNotFoundException("Пользователь не найден");
+        }
+        User user = byEmail.get();
```
всю конструкцию можено заменить на orElseThrow

OlgaLugacheva commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -53,14 +53,19 @@ public List<Ads> getAllAds() {
         return saveAds;
+        } catch (IOException e){
+            log.error("Ошибка при сохранении объявления", e);
+            throw new RuntimeException("Ошибка при сохранении объявления", e);
```
опять

OlgaLugacheva commented on src/test/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,121 @@
+//package ru.skypro.homework.controller;
+//
```
наверно, UserControllerTest и , наверно, стоит его привести в порядок и работоспособное состояние, раз он тут есть

AlexandraGlinka committed:
UserControllerTest / getAuthUserInfoTest

AlexandraGlinka committed:
UserServiceImplTest / setPasswordTest

korzhuk-xeniya committed:
добавлен README

korzhuk-xeniya committed:
добавлены JavaDoc в AdsController

korzhuk-xeniya committed:
добавлены JavaDoc в CommentServiceImpl

korzhuk-xeniya committed:
добавлены JavaDoc в AdsServiceImpl

el-kg committed:
-В AdsController поправлен выброс исключения в методе AddAd (заменен на кастомный AdsNotFoundException)

-В ImageController поправлен выброс исключения в методе transferImageToResponse (заменен на кастомный ImageNotFoundException)

-Убрал лишнее описание из GlobalExceptionHandler

-Поправлен выброс исключения в AdsServiceImpl  в методе saveAd (заменен на кастомный AdsNotFoundException)

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/3-feature-comment-tests-kxeny' into 3-feature-step-3

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/3-feature-step-3' into 3-feature-step-3

AlexandraGlinka committed:
удален лишний код в пользователе

AlexandraGlinka committed:
Merge remote-tracking branch 'origin/3-feature-pre-diplom-aglinka' into 3-feature-step-3

AlexandraGlinka committed:
UserControllerTest / updateAuthUserInfoTest и updateAvatarTest

AlexandraGlinka committed:
Merge remote-tracking branch 'origin/3-feature-pre-diplom-aglinka' into 3-feature-step-3

el-kg committed:
-В SecurityUserService поправлен выброс исключения в методе loadUserByUsername (конструкция заменена на orElseThrow)

AlexandraGlinka committed:
небольшие замечания от Оли с защиты
